pkg/node/node.go

```diff
@@ -241,15 +244,13 @@ func (n *Node) publishSignerPublicKey(
 		publicKey,
 	)
 
-	monitoringAbort := make(chan interface{})
-	go n.monitorKeepPublicKeySubmission(monitoringAbort, keepAddress)
-
 	err := n.ethereumChain.SubmitKeepPublicKey(keepAddress, publicKey)
 	if err != nil {
-		close(monitoringAbort)
 		return fmt.Errorf("failed to submit public key: [%v]", err)
```

Comment:
I don't think so. `SubmitKeepPublicKey` retries on error under the hood. If we have an error here I don't see any sense to retry from `monitorKeepPublicKeySubmission`. Also, we can't just log the error and rely on asynchronous `monitorKeepPublicKeySubmission` because we have to return the `signer` (or not) and properly handle this keep in the client.

Reply:
If this `SubmitKeepPublicKey` fails, we do not return an error (my proposition), and `monitorKeepPublicKeySubmission` does not eventually succeed with publishing the key to the chain, we will return a `signer` that will be properly handled in the client but will never be asked to sign anything *unless* the transaction will be manually re-submitted from outside of the client by one of the operators. Something we have already seen once or twice by the way.

Now, assuming we leave this code as-is and the first `SubmitKeepPublicKey` does not fail but the transaction will not be mined for whatever the reason - keep in mind no error returned from this function does not guarantee the key to end on the chain - and `monitorKeepPublicKeySubmission` does not eventually succeed with publishing the key to the chain, we are going to have exactly the same situation, right?

What feels wrong to me: the fact this function does not return an error does not guarantee the signature will land on the chain and that the `signer` will ever be asked for providing a signature. Also, we cancel the process on error returned from `SubmitKeepPublicKey` but only for the first invocation. We do not do it for `SubmigKeepPublicKey` inside `monitorKeepPublicKeySubmission`. Maybe there is a valid reason for that but I do not see it yet. Why treat those two identical invocations differently? 

> If we have an error here I don't see any sense to retry from monitorKeepPublicKeySubmission

Let me reverse this statement: if we have an error in `SubmigKeepPublicKey` in `monitorKeepPublicKeySubmission` what is a sense of continuing this loop instead of just returning? 

> `SubmitKeepPublicKey` retries on error under the hood.

That's a very good point. So maybe instead of modifying anything here, we should exit on error returned from `SubmitKeepPublicKey` in `monitorPublicKeySubmission`?

Reply:
More thoughts on this matter: retries in `SubmitKeepPublicKey` helps to overcome infrastructure-level issues like problem with Ethereum client not fully synced with the most recent state.

What we are trying to protect against here are reorgs. And protection about a reorg is all about noting the time the state has changed, then waiting for N blocks, confirming the state, and either repeating an action or leaving. Having this in mind, if feels to me that returning from `monitorKeepPublicKeySubmission` on `SubmitKeepPublicKey`  error is the cleanest option. Thoughts?